helix-term/src/commands.rs

```diff
@@ -4390,28 +4390,39 @@ fn format_selections(cx: &mut Context) {
 fn join_selections_impl(cx: &mut Context, select_space: bool) {
     use movement::skip_while;
     let (view, doc) = current!(cx.editor);
-    let text = doc.text();
-    let slice = doc.text().slice(..);
 
     let mut changes = Vec::new();
-    let fragment = Tendril::from(" ");
 
     for selection in doc.selection(view.id) {
-        let (start, mut end) = selection.line_range(slice);
-        if start == end {
-            end = (end + 1).min(text.len_lines() - 1);
-        }
-        let lines = start..end;
+        let slice = doc.text().slice(..);
+        let text = doc.text();
 
-        changes.reserve(lines.len());
+        let selected_lines = {
+            let (start, mut end) = selection.line_range(slice);
+            if start == end {
+                end = (end + 1).min(text.len_lines() - 1);
+            }
+            start..end
+        };
 
```

Comment:
Could you reset the unnecessary changes like variable renames and moving the `slice` and `text` bindings? Inlining `Tendril::from(" ")` is ok since that will avoid some allocations but the rest is just cosmetic

Reply:
updated. I'd like to have this block at least:
```rust
        let lines = {
            let (start, mut end) = selection.line_range(slice);
            if start == end {
                end = (end + 1).min(text.len_lines() - 1);
            }
            start..end
        };
```
makes it more readable, in my opinion, since we are seeing that we just need `stard` and `end` for `lines` but I can remove it if you insist.

Reply:
I think this kind of extra indention ist actually very unreadable and clutterwd. Please reverse the Extra nestong jere and also remove the unnecssary nestong in the cide you added.

Reply:
Yeah in general I'm not a big fan of the `let var = { .. }` syntax unless we're avoiding some really nasty scoping issues. Readability / cosmetic changes like this are always debatable and they inflate the diff (make review harder and might introduce unnecessary conflicts with other PRs) so we try to avoid them

Reply:
reverted